### PR TITLE
Add TLE fused AttnRes forward kernel

### PR DESCRIPTION
**Pull request by lauuyx-png**
## Summary

Add a forward-only TLE implementation of Kimi K3 Attention Residuals (`fused_attnres`).

- Fuse per-source RMSNorm scoring, depth/source softmax, residual aggregation, and optional output RMSNorm into one kernel.
- Use a one-read online softmax so each residual tile is loaded once for both scoring and mixing.
- Pass residual tensors as a padded pointer tuple instead of building and copying a CPU pointer table.
- Autotune source block size, warp count, and pipeline stages for the Kimi K3 target shapes.
- Add a float32 PyTorch reference, correctness tests, and an opt-in preallocated-kernel benchmark.

The API is intended for inference. It supports FP16/BF16, up to 16 residual sources, and hidden sizes up to 8192; the primary target is Kimi K3 (`D=7168`, up to 9 sources). It requires a Triton build with `triton.experimental.tle` and does not provide an autograd formula.

## Reproducible baseline and compiler

- FLA baseline: [`fla-org/flash-linear-attention@5aea42b7740f9968f6418c6c60b78ea785ce6140`](https://github.com/fla-org/flash-linear-attention/commit/5aea42b7740f9968f6418c6c60b78ea785ce6140)
- Baseline implementation: `fla/ops/attnres/fused.py::attnres_fwd_kernel`
- Baseline file blob: `d6a4e59c70472e65efd6c5adf9d22cca9e254c8b`
- FlagTree: [`flagos-ai/FlagTree@aaa420f9366440e18bd83a0391c2d2dbcc5e0b81`](https://github.com/flagos-ai/FlagTree/commit/aaa420f9366440e18bd83a0391c2d2dbcc5e0b81), branch `triton_v3.6.x`, Triton `3.6.0`
- FlagAttention commit: `5480847`

The benchmark was rerun from a clean detached FLA worktree at the public commit above. The previous local-only provenance and its results have been removed.

## H100 preallocated-kernel comparison

Both implementations receive identical BF16 inputs with `D=7168`. Pointer structures and every output/stat tensor are allocated before timing. The timed call directly launches each existing forward kernel, excluding Python public wrappers, tensor allocation, residual pointer-table construction/H2D, and postprocessing.

`triton.testing.do_bench` is called with `warmup=1000`, `rep=100`, and explicit `return_mode="median"`. `warmup` and `rep` are millisecond time budgets. The `1000/100` budgets match the FlagGems kernel benchmark defaults. Each provider is measured for seven rounds in alternating order; the table reports the median of the seven per-round medians. L2 cache clearing performed by `do_bench` remains enabled.

The public FLA kernel writes `p`, `rstd`, and `score_mean` because those statistics are intrinsic to its training-capable forward kernel. The FlagAttention inference kernel does not write backward statistics. This table compares the existing forward kernels and does not claim identical internal auxiliary-output semantics.

Environment: NVIDIA H100 80GB HBM3, PyTorch `2.10.0+cu128`, FlagTree/Triton `3.6.0`.

### Without output RMSNorm

| sources | rows | FLA kernel (ms) | FlagAttention kernel (ms) | FLA / FlagAttention |
|---:|---:|---:|---:|---:|
| 2 | 1 | 0.025536 | 0.007776 | 3.284x |
| 5 | 128 | 0.029664 | 0.013856 | 2.141x |
| 5 | 1024 | 0.141168 | 0.044192 | 3.194x |
| 9 | 128 | 0.035136 | 0.017920 | 1.961x |
| 9 | 8192 | 1.244800 | 0.450880 | 2.761x |

### With fused output RMSNorm

| sources | rows | FLA kernel (ms) | FlagAttention kernel (ms) | FLA / FlagAttention |
|---:|---:|---:|---:|---:|
| 2 | 1 | 0.029760 | 0.008256 | 3.605x |
| 5 | 128 | 0.033184 | 0.014592 | 2.274x |
| 5 | 1024 | 0.142256 | 0.047264 | 3.010x |
| 9 | 128 | 0.038304 | 0.018848 | 2.032x |
| 9 | 8192 | 1.376048 | 0.466496 | 2.950x |

Across the tested Kimi K3 shapes, the preallocated TLE forward kernel is `1.961x-3.605x` faster than the existing FLA forward kernel.

## Correctness

The default pytest comparison checks output and optional source weights against the float32 PyTorch reference. Coverage includes 1, 5, and 9 sources; `D=7168` Kimi K3 shapes; FP16 and BF16; output RMSNorm on/off; and the API path without returned weights.

```text
6 passed, 10 skipped
```

Each opt-in benchmark case first compares the two public API outputs and then compares the preallocated kernel outputs before timing:

```text
10 passed, 6 deselected
```

## Validation

```bash
FLA_BASE=/share/project/tle/lau/flash-linear-attention-attnres-baseline
FLAGATTN=/share/project/tle/lau/FlagAttention-attnres
FLAGTREE=/share/project/tle/FlagTree/build/lib.linux-x86_64-cpython-312

git -C "$FLA_BASE" rev-parse HEAD
# 5aea42b7740f9968f6418c6c60b78ea785ce6140

git -C /share/project/tle/FlagTree rev-parse HEAD
# aaa420f9366440e18bd83a0391c2d2dbcc5e0b81

PYTHONPATH="$FLAGATTN/src:$FLA_BASE:$FLAGTREE" \
/root/miniconda3/envs/tle/bin/python -m pytest \
  "$FLAGATTN/tests/flag_attn/test_attnres.py" -q

FLAG_ATTN_RUN_EXTERNAL_BENCHMARKS=1 \
PYTHONPATH="$FLAGATTN/src:$FLA_BASE:$FLAGTREE" \
/root/miniconda3/envs/tle/bin/python -m pytest \
  "$FLAGATTN/tests/flag_attn/test_attnres.py" -k kernel_benchmark -q -s
```

### CODE REVIEW

File: tests/flag_attn/test_attnres.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        rep=100,
+    )
+    flag_attn_ms = triton.testing.do_bench(
+        lambda: flag_attn.fused_attnres(
+            query, residuals, rms_weight, output_rms_weight, scale=scale
+        ),
+        warmup=25,
+        rep=100,
+    )
+    speedup = fla_ms / flag_attn_ms
```
Comment (huangyiqun) [inline tests/flag_attn/test_attnres.py:275]: It looks like the end-to-end latency of `fla_fused_attnres` & `flag_attn.fused_attnres` was tested here.

**Review by huangyiqun — DISMISSED** [review]
